src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -345,7 +344,8 @@ end = struct
     in
     go [] [] ts
 
-  let check_completed_works scan_state
+  (* TODO? *)
+  let _check_completed_works scan_state
```

Comment:
It seems like a bug that this is unused, cc @deepthiskumar @psteckler

Reply:
Yeah, I wanted to check on that. This was only used in the verified staged ledger code, but we need to make sure this check still exists when we apply the staged ledger diff.

Reply:
This was moved out of `apply` to remove all the `Deferred`s. Can we not have Staged_ledger_diff.Verified.t only in `Staged_ledger.t` (or maybe `Staged_ledger_diff.With_verified_transaction_snarks.t` if the terminology is misleading) and perform the check in `Transition_frontier` before applying?

Reply:
Yes, that was the key piece of code to make unverified work become verified work.